### PR TITLE
LoginProducer unit tests for Github and Gitlab providers.

### PR DESCRIPTION
**Pull request by nmonokov**
PR for #321

### CODE REVIEW

**Comment by charlesmike** [comment]
@nmonokov thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2021-01-30T02:44:03.241218``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: src/test/java/com/selfxdsd/selfweb/LoginProducerTestCase.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                .thenReturn("github");
+        final OAuth2AuthorizedClientService clientService =
+                Mockito.mock(OAuth2AuthorizedClientService.class);
+
+        LoginProducer producer = new LoginProducer(core);
+        producer.login(clientService);
+
+        Mockito.verify(core, Mockito.only()).login(
+                Mockito.any(Login.class)
+        );
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/LoginProducerTestCase.java:49]: @nmonokov Can you also add an assertion that the returned User is actually the ``authenticated`` User declared above?

Something like:

```java
MatcherAssert.assertThat(
    producer.login(clientService),
    Matchers.is(authenticated)
)
```

And the same for the test bellow.

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@nmonokov looks good, thanks! I just have one comment/question.

**Comment by amihaiemil** [comment]
@nmonokov also, please always provide a description to the PR, linking the ticket that you are solving.

**Comment by amihaiemil** [comment]
@rultor merge it please

**Comment by amihaiemil** [comment]
@rultor merge it please

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**